### ISSUE TITLE
openssl req -newkey failing: Bug or documentation issue?

### ISSUE THREAD

**Issue by baentsch** [openssl/openssl#15569]
On the latest `master`, this command fails:

```
> ./apps/openssl req -x509 -new -newkey rsa -keyout CA.key -out CA.crt -nodes -subj '/CN=test CA' -days 365 -config apps/openssl.cnf 
Private key length too short, needs to be at least 512 bits, not -1.
```

This is part of our regular test suite and always worked OK before the latest `git pull`. If this is intentional, please consider updating the documentation for `req -newkey` which currently still reads:
```
 -newkey val           Generate new key with [<alg>:]<nbits> or <alg>[:<file>] or param:<file>
```
indicating our use (only specifying an "alg" but no "nbits") is still permissible.

If explicitly specifying the key length, the command works again:
```
./apps/openssl req -x509 -new -newkey rsa:2048 -keyout CA.key -out CA.crt -nodes -subj '/CN=test CA' -days 365 -config apps/openssl.cnf
```

This has been tested on commit `147ed5f9def86840c9f6ba512e63a890d58ac1d6` on Linux (locally and in CI environments).

**Comment by t8m** [comment]
Fix in #15582. Please try it.

**Comment by baentsch** [comment]
Just tried it: Fixes the issue: Thanks!
